parcels/include/parcels.h

```diff
@@ -954,6 +954,265 @@ static inline StatusCode temporal_interpolationUVW_c_grid(type_coord x, type_coo
   }
 }
 
+static inline StatusCode calculate_slip_conditions_2D(double xsi, double eta, float dataU[2][2],
+                                                      float dataV[2][2], float dataW[2][2],
+                                                      float *u, float *v, float *w, int interp_method, int withW)
+{
+      float f_u = 1, f_v = 1, f_w = 1;
+      if ((is_zero_flt(dataU[0][0])) && (is_zero_flt(dataU[0][1])) &&
+          (is_zero_flt(dataV[0][0])) && (is_zero_flt(dataV[0][1])) && eta > 0.){
```

Comment:
good that you used our numerically-stable 0-check.

Reply:
Thanks!